app.py

```diff
@@ -63,23 +68,52 @@ def _event_producer(queue: Queue, build_watcher_namespace: str):
     openshift = OpenShift()
     v1_build = openshift.ocp_client.resources.get(api_version="v1", kind="Build")
     for event in v1_build.watch(namespace=build_watcher_namespace):
+        event_name = event["object"].metadata.name
+        build_reference = {"build_log_reference": None, "base_input_reference": None, "output_reference": None}
         if event["object"].status.phase != "Complete":
             _LOGGER.debug(
-                "Ignoring build event for %r - not completed phase %r",
-                event["object"].metadata.name,
-                event["object"].status.phase,
+                "Ignoring build event for %r - not completed phase %r", event_name, event["object"].status.phase
             )
             continue
+        elif event["object"].status.phase == "Failed":
+            _LOGGER.debug(
+                "Submitting base_image and build_log as build event for %r - the phase is %r",
+                event_name,
+                event["object"].status.phase,
+            )
+            build_reference["base_input_reference"] = event["object"].spec.strategy.sourceStrategy.from.name
+            build_log = get_build_log(build_id=event_name, namespace=event["object"].metadata.namespace)
+            build_reference["build_log_reference"] = _buildlog_metadata(event["object"].metadata.selfLink, build_log)
+            _LOGGER.info(build_reference["base_input_reference"], build_reference["build_log_reference"])
+            _LOGGER.info(
+                "Queueing %r based on build event %r for further processing",
+                build_reference["build_log_reference"],
+                event_name,
+            )
+            queue.put(build_reference)
 
         _LOGGER.debug("New build event: %s", str(event))
-        event_name = event["object"].metadata.name
-        output_reference = event["object"].status.outputDockerImageReference
-        _LOGGER.info("Queueing %r based on build event %r for further processing", output_reference, event_name)
-        queue.put(output_reference)
+        build_reference["output_reference"] = event["object"].status.outputDockerImageReference
+        build_reference["base_input_reference"] = event["object"].spec.strategy.sourceStrategy.from.name
+        build_log = get_build_log(build_id=event["object"].metadata.name, namespace=event["object"].metadata.namespace)
+        build_reference["build_log_reference"] = _buildlog_metadata(event["object"].metadata.selfLink, build_log)
+        _LOGGER.info(
+            build_reference["output_reference"],
+            build_reference["base_input_reference"],
+            build_reference["build_log_reference"],
+        )
+        _LOGGER.info(
+            "Queueing %r based on build event %r for further processing",
+            build_reference["build_log_reference"],
+            event_name,
+        )
+        queue.put(build_reference)
```

Comment:
Does it mean there are two events to handle? Two times queue put (line 93).

Reply:
Yes, there would be two events, one if the build failed then only build logs and base image would be submitted and another when the build succeeds then output image, build logs and base image would be submitted for analysis.